src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -171,6 +179,45 @@ class BuiltGoPackage:
     import_paths_to_pkg_a_files: FrozenDict[str, str]
 
 
+@dataclass
+@frozen_after_init
+class EmbedConfig:
+    """Configuration passed to the compiler to configure file embedding. The compiler relies
+    entirely on the caller to map file patterns to actual filesystem paths.
+
+    patterns: Maps a pattern to a list of _relative_ file paths to use for that pattern. These relative file paths
+    are "virtual" because they will be resolved to actual filesystem paths by consulting the `files` dictionary.
+
+    files: Maps a "virtual" _relative_ file path used as a value in the `patterns` dictionary to the actual
+    filesystem path with the file's content.
+    """
```

Comment:
FYI you can move this to `__init__` so that you can use `:param files:`. I think PyCharm at least knows how to handle this better than the classdoc you have now

Reply:
Done.